sale_order_line_cancel/models/sale_order.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    def action_draft(self):
+        res = super().action_draft()
+        orders = self.filtered(lambda s: s.state == "draft")
+        orders.order_line.write({"product_qty_canceled": 0})
+        return res
+
+    def _action_cancel(self):
+        orders = self.filtered(lambda s: s.state != "cancel")
+        orders_with_picking = orders.filtered("picking_ids")
```

Comment:
Maybe this is better as in `sale_stock` they use `qty_delivered_method`:
```suggestion
        orders_with_picking = orders.filtered(lambda o: any(line.qty_delivered_method == 'stock_move' for line in o.order_line)):
```